lib/Runtime/Base/ThreadContext.cpp

```diff
@@ -1035,37 +1035,18 @@ ThreadContext::UncheckedAddPropertyId(JsUtil::CharacterBuffer<WCHAR> const& prop
 
     int length = propertyName.GetLength();
     uint bytelength = sizeof(char16) * length;
-
-    uint32 indexVal = 0;
-
-    // Symbol properties cannot be numeric since their description is not to be used!
-    bool isNumeric = !isSymbol && Js::PropertyRecord::IsPropertyNameNumeric(propertyName.GetBuffer(), propertyName.GetLength(), &indexVal);
-
-    uint hash = JsUtil::CharacterBuffer<WCHAR>::StaticGetHashCode(propertyName.GetBuffer(), propertyName.GetLength());
-
-    size_t allocLength = bytelength + sizeof(char16) + (isNumeric ? sizeof(uint32) : 0);
+    size_t allocLength = bytelength + sizeof(char16) + ( (!isSymbol && length <= 10 && length > 0) ? sizeof(uint32) : 0);
```

Comment:
well, don't we already know that for length 2, 3 and 10 we will allocate extra 16 bytes (at least for x64). Why can't we add extra condition here to return `sizeof(uint32)` vs. sizeof(uint16)`?

---
In reply to: [130432408](https://github.com/Microsoft/ChakraCore/pull/3454#discussion_r130432408) [](ancestors = 130432408)